### DISCUSSION TITLE
Export from already existing MassCode?

### DISCUSSION THREAD

**Discussion by kaufsim** [massCodeIO/massCode#3] [Q&A] [answered]
Hi

I was using Masscode until now already and I love it! Will the new MassCode reuse the same database or are you planning to move to a new solution? If yes, will there be a possibility to export / import the snippets?

Thanks and keep up the great work :-)

**Comment by antonreshetov** [comment, marked as answer]
@kaufsim Of course the migration from v1 is provided and already implemented.

**Comment by cacan** [comment]
Hi @antonreshetov , great tool... 

Using v2.0.0 beta1, but from : https://github.com/antonreshetov/massCode

Is it compatible with this new v2? 

![image](https://user-images.githubusercontent.com/406493/162585581-b1103df8-908d-4830-9827-e7f42fd9a414.png)

**Reply by antonreshetov** [reply]
Yes, I need to explain a little.  This is a brand new application written from scratch.  Since v2 was never released in the old repository, I decided to leave 2 in versioning for the new application.  If I took the number 3 for a new application, then the question would probably be where is version 2?  These are the thoughts 😂

**Comment by cacan** [comment]
Thanks... But sill - is this new release the same as this one: https://github.com/antonreshetov/massCode/releases/tag/v2.0.0-beta.1

Or the new release is completely rewritten [this is how I got it]? 

If it's rewritten, what would happen if I install new v2 over this "old" v2 beta: https://github.com/antonreshetov/massCode/releases/tag/v2.0.0-beta.1

**Reply by antonreshetov** [reply]
This repository contains a new version of the application. The old version https://github.com/antonreshetov/massCode/releases/tag/v2.0.0-beta.1 can be forgotten.

**Reply by cacan** [reply]
Understand that, but will the new app import "old v2" data [as it was in a different format from v1]?

**Reply by antonreshetov** [reply]
Unfortunately not, migration only from v1. Maybe later.

**Reply by cacan** [reply]
So, those who were too eager are now stuck with "old v2"? :(

**Reply by antonreshetov** [reply]
This sometimes happens, especially since the beta stage is not inherently stable. You can contribute and implement the migration.

**Reply by v-c0de** [reply]
Yeah, unfortunately I'm in the same boat. I might have to reinstall the 'old v2' and then manually migrate them across. I tried looking at the JSON definition to see if i can bulk load them, but I don't want to ruin the id setup. Thankfully i didn't fully migrate everything from my main snippets manager!

**Reply by antonreshetov** [reply]
@v-c0de snippet or folder ids it's simple `string` generate by [nanoid](https://github.com/ai/nanoid). You can use any random `string` what you want.

**Reply by eelke** [reply]
@antonreshetov First of all, thanks for all your work. When it comes to versioning though, maybe you could take a look into [semantic versioning](https://semver.org/) . I also just upgraded from the 2.0.0 beta to 2.5 and to my large surprise all my Snippets are missing. SemVer dictactes that upgrading between minor versions should never introduce any breaking changes, only deprecations and new features. The following major release then can remove these deprecations. So this new version should have been 3.0 I guess. And personally I will now revert to the old beta.

**Reply by antonreshetov** [reply]
@eelke I know about semver, so it's okay with the versions. Read again what I wrote about this case.

**Reply by eelke** [reply]
@antonreshetov Maybe I misunderstood, or something else is going wrong. But upgrading from a 2.0.0 beta to 2.5.0 made all my snippets disappear. From the perspective of a user that means: either this is a bug or a breaking change. And now that I've reinstalled 2.0.0-beta1 I don't have any snippets showing up anymore either. What a bummer! All snippets seem fine and available inside the ~/massCode/db.json database.

**Reply by antonreshetov** [reply]
@eelke You tell me thank you, but it sounds like just empty words, because afterwards you criticize and do not offer solutions. For example a contribution to solve a possible bug. You use an open-source application and for some reason you think someone owes you as if you paid for it. You haven't even given the project a star. Obviously consumer behavior. 

P.S. Advice not to use beta versions as stable versions. 
P.P.S Old versions of the app have a backup so you can restore snippets.

**Reply by antonreshetov** [reply]
Well, I don't have the ability to buy an Apple Developer Program membership. It's kind of a shame to point that out. 

I'll just leave it here - https://semver.org/#spec-item-9
